cli/ballerina-cli/src/main/java/io/ballerina/cli/diagnostics/AnnotateDiagnostics.java

```diff
@@ -0,0 +1,286 @@
+        boolean isMultiline = diagnostic.message().contains(NEW_LINE);
+        boolean isCodeNotNull = code != null;
+        String formatString = getColoredString("%s", color, colorEnabled) + "%s" +
+                (isCodeNotNull ? (isMultiline ? NEW_LINE + "(%s)" : " (%s)") : "");
+
+        return String.format(formatString, severityString, message, isCodeNotNull ? code : "");
+    }
+
+    private static DiagnosticAnnotation getDiagnosticAnnotation(Document document, Location location,
+                                                                DiagnosticSeverity severity, int terminalWidth,
+                                                                boolean colorEnabled) {
+        TextDocument textDocument = document.textDocument();
+        LocationDetails locationDetails = getLocationDetails(location);
+        boolean isMultiline = locationDetails.startLine != locationDetails.endLine;
+        int length = isMultiline ? textDocument.line(locationDetails.startLine).length() - locationDetails.startOffset :
+                locationDetails.endOffset - locationDetails.startOffset;
+
+        return new DiagnosticAnnotation(
+                getLines(textDocument, locationDetails.startLine, locationDetails.endLine),
+                locationDetails.startOffset,
+                length == 0 ? 1 : length,
+                isMultiline,
+                locationDetails.endOffset,
+                locationDetails.startLine + 1,
+                severity,
+                DiagnosticAnnotation.DiagnosticAnnotationType.REGULAR,
+                terminalWidth, colorEnabled);
+    }
+
+    private static DiagnosticAnnotation getSyntaxDiagnosticAnnotation(Document document,
+                                                                      PackageDiagnostic packageDiagnostic,
+                                                                      int diagnosticCode, int terminalWidth,
+                                                                      boolean colorEnabled) {
+        TextDocument textDocument = document.textDocument();
+        Location location = packageDiagnostic.location();
+        LocationDetails locationDetails = getLocationDetails(location);
+        int padding = 0;
+        String color = SEVERITY_COLORS.get(DiagnosticSeverity.ERROR);
+
+        if (diagnosticCode < MISSING_TOKEN_KEYWORD_CODE_THRESHOLD) {
+            return getMissingTokenAnnotation(packageDiagnostic, textDocument, locationDetails, color, colorEnabled,
+                    terminalWidth, padding);
+        }
+
+        if (diagnosticCode == INVALID_TOKEN_CODE) {
+            return getInvalidTokenAnnotation(textDocument, locationDetails, color, colorEnabled, terminalWidth);
+        }
+        return getDiagnosticAnnotation(document, location, DiagnosticSeverity.ERROR, terminalWidth,
+                colorEnabled);
+    }
+
+    private static DiagnosticAnnotation getMissingTokenAnnotation(PackageDiagnostic packageDiagnostic,
```

Comment:
Same as above, check places where you are passing the `boolean colorEnabled, int terminalWidth, int padding`, paramters